pkg/pipeline/decode/decode_protobuf.go

```diff
@@ -0,0 +1,85 @@
+package decode
+
+import (
+	"fmt"
+	"net"
+	"time"
+
+	"github.com/netobserv/flowlogs-pipeline/pkg/config"
+	"github.com/netobserv/netobserv-agent/pkg/pbflow"
+	"github.com/sirupsen/logrus"
+)
+
+var pflog = logrus.WithField("component", "Protobuf")
+
+// Protobuf decodes protobuf flow records definitions, as forwarded by
+// ingest.NetObservAgent, into a Generic Map that follows the same naming conventions
+// as the IPFIX flows from ingest.IngestCollector
+type Protobuf struct {
+}
+
+func NewProtobuf() (Decoder, error) {
+	return &Protobuf{}, nil
+}
+
+// Decode decodes input strings to a list of flow entries
+func (c *Protobuf) Decode(in []interface{}) []config.GenericMap {
+	if len(in) == 0 {
+		pflog.Warn("empty input. Skipping")
+		return []config.GenericMap{}
+	}
+	pb, ok := in[0].(*pbflow.Records)
+	if !ok {
+		pflog.WithField("type", fmt.Sprintf("%T", pb)).
+			Warn("expecting input to be *pbflow.Records. Skipping")
+	}
+	out := make([]config.GenericMap, 0, len(pb.Entries))
+	for _, entry := range pb.Entries {
+		out = append(out, pbFlowToMap(entry))
+	}
+	return out
+}
+
+func pbFlowToMap(flow *pbflow.Record) config.GenericMap {
+	if flow == nil {
+		return config.GenericMap{}
+	}
+	out := config.GenericMap{
+		"FlowDirection": int(flow.Direction.Number()),
+		"Bytes":         flow.Bytes,
+		"SrcAddr":       ipToStr(flow.Network.GetSrcAddr()),
+		"DstAddr":       ipToStr(flow.Network.GetDstAddr()),
+		"SrcMac":        macToStr(flow.DataLink.GetSrcMac()),
+		"DstMac":        macToStr(flow.DataLink.GetDstMac()),
+		"SrcPort":       flow.Transport.GetSrcPort(),
+		"DstPort":       flow.Transport.GetDstPort(),
+		"Etype":         flow.EthProtocol,
+		"Packets":       flow.Packets,
+		"Proto":         flow.Transport.GetProtocol(),
+		"TimeFlowStart": flow.TimeFlowStart.GetSeconds(),
+		"TimeFlowEnd":   flow.TimeFlowEnd.GetSeconds(),
+		"TimeReceived":  time.Now().Unix(),
+		"Interface":     flow.Interface,
+	}
+	return out
+}
+
+func ipToStr(ip *pbflow.IP) string {
```

Comment:
If `ipToStr` and `macToStr` are generic, maybe move to some utility or common file